### ISSUE TITLE
inconsistency in argument order between regression and classification's  build_tree

### ISSUE THREAD

**Issue by Eight1911** [bensadeghi/DecisionTree.jl#76]
regression's `build_tree` has the following signature
```julia
function build_tree(
        labels             :: Vector{T},
        features           :: Matrix{S},
        min_samples_leaf    = 5,
        n_subfeatures       = 0,
        max_depth           = -1,
        min_samples_split   = 2,
        min_purity_increase = 0.0;
        rng                 = Random.GLOBAL_RNG) where {S, T <: Float64}
```
while classification's `build_tree` has the following signature
```julia
function build_tree(
        labels              :: Vector{T},
        features            :: Matrix{S},
        n_subfeatures        = 0,
        max_depth            = -1,
        min_samples_leaf     = 1,
        min_samples_split    = 2,
        min_purity_increase  = 0.0;
        rng                  = Random.GLOBAL_RNG) where {S, T}
```
The third argument, `min_samples_leaf` in regression should be the fifth argument for argument consistency. This should be changed since it may cause silent bugs for users.

**Comment by bensadeghi** [comment]
@Eight1911 Then we should have consistent argument order for ```build_forest``` as well.
For regression, we can move ```min_samples_leaf``` to the end.
And for classification, we can add ```min_samples_leaf``` to the end

**Comment by bensadeghi** [comment]
@Eight1911 Are you resolving this as part of the next push, or should I take it?

**Comment by Eight1911** [comment]
this will be resolved in the next push

**Comment by bensadeghi** [comment]
@Eight1911 You rock! Thanks

**Comment by bensadeghi** [comment]
Resolved with PR #82
